language-support/ts/daml-ledger/index.ts

```diff
@@ -25,6 +25,52 @@ export type Event<T extends object, K = unknown, I extends string = string> =
   | { created: CreateEvent<T, K, I> }
   | { archived: ArchiveEvent<T, I> }
 
+/**
+ * `CreateEventUnknown` is equivalent to `CreateEvent<object>` with the added
+ * benefit that the type checker understands it better.
+ */
+export interface CreateEventUnknown extends CreateEvent<object> {
+  contractId: ContractId<object>;
+}
+
+/**
+ * `ArchiveEventUnknown` is equivalent to `ArchiveEvent<object>` with the added
+ * benefit that the type checker understands it better.
+ */
+export interface ArchiveEventUnknown extends ArchiveEvent<object> {
+  contractId: ContractId<object>;
+}
```

Comment:
I'm not sure I understand why these specializations are truly required; does changing usages of `CreateEvent<object>` to `CreateEvent<any>` make things better?

Reply:
Tbh, I don't fully understand it either. It seems like a shortcoming in the TS type checker to me. I've tried it with `CreateEvent<unknown>` and that didn't help either. ☹️